### DISCUSSION TITLE
Show gutter diff for 'src' VCS

### DISCUSSION THREAD

**Discussion by Eloitor** [helix-editor/helix#9820] [General]
The 'src' VCS is a modern UI for RCS. https://www.catb.org/~esr/src/

Since it is not a very popular VCS, maybe it does not make sense to add build in support for it, but a diff for a file can be obtained using `src diff file_name`.

Maybe the gutter could be configured to use a user defined command to get a diff...

**Comment by pascalkuthe** [comment]
This is plugins territory.  I never heard of this vcs before.  We only include git in core because it's ubiquitous. 

We also don't shell out to compute a diff but instead use our own git implementation to retrieve the file at head and then run a diff against that on the fly.

So full support for a vcs needs a way to properly check out a file and in the future vcs paths also need to be watched for changes

**Comment by Eloitor** [comment]
Ok thanks for the response... 
Maybe the command `src cat file_name` could be used which gets the last commited version of the file.
And let helix compute the diffs.
